pkgs/development/libraries/libftdi/1.x.nix

```diff
@@ -1,47 +1,71 @@
-{ lib, stdenv, fetchurl, cmake, pkg-config, libusb1, libconfuse
-, cppSupport ? true, boost ? null
-, pythonSupport ? true, python3 ? null, swig ? null
-, docSupport ? true, doxygen ? null
+{ lib
+, stdenv
+, fetchgit
+, cmake
+, pkg-config
+, libusb1
+, libconfuse
+, cppSupport ? true
+, boost ? null
+, pythonSupport ? true
+, python3 ? null
+, swig ? null
+, docSupport ? true
+, doxygen ? null
 }:
 
 assert cppSupport -> boost != null;
 assert pythonSupport -> python3 != null && swig != null;
 assert docSupport -> doxygen != null;
-
+let
+  inherit (lib) optionals;
+  onOff = a: if a then "ON" else "OFF";
```

Comment:
There should be a lib function which does exactly this for cmake. I think it is call with or something.

Reply:
Nothing in lib.strings is really useful for this since all of the `{with,without,enable}Feature(As)` functions there return a Autoconf-styled switch (`--*-<feature>=<value>`) instead of CMake-styled one (`-D<FEATURE>=<VALUE>`). It's so trivial & common that it really *should* be in some library, but so far it's reimplemented all over the place…

https://github.com/NixOS/nixpkgs/blob/d600f006643e074c2ef1d72e462e218b647a096c/pkgs/development/libraries/stxxl/default.nix#L6

Reply:
I'll mark this as resolved, unless you have a more specific suggestion for a library function.